File: backend/src/main/java/com/shoptourr/api/v1/dto/map/MapDtos.java
package com.shoptourr.api.v1.dto.map;

import com.shoptourr.api.v1.dto.common.CommonDtos.MoneyDto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

/**
 * Map / route screen. v1 may return stops derived from purchase places
 * (geocoded server-side); lat/lng nullable until geocode succeeds.
 */
public final class MapDtos {

    private MapDtos() {}

    public record GeoPointDto(
            BigDecimal lat,
            BigDecimal lng
    ) {}

    public record RouteStopDto(
            UUID id,
            String title,
            String place,
            LocalDate date,
            MoneyDto amountSpentHere,
            GeoPointDto point,
            int orderIndex
    ) {}

    public record TripRouteDto(
            UUID tripId,
            int stopCount,
            /** Meters; null if unknown. */
            BigDecimal distanceMeters,
            List<RouteStopDto> stops,
            /** Ordered polyline if available. */
            List<GeoPointDto> path
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/data/remote/dto/map/MapDtos.kt
package com.example.shoptourr.data.remote.dto.map

import com.example.shoptourr.data.remote.dto.common.MoneyDto

data class GeoPointDto(
    val lat: String,
    val lng: String,
)

data class RouteStopDto(
    val id: String,
    val title: String,
    val place: String? = null,
    val date: String? = null,
    val amountSpentHere: MoneyDto? = null,
    val point: GeoPointDto? = null,
    val orderIndex: Int,
)

data class TripRouteDto(
    val tripId: String,
    val stopCount: Int,
    val distanceMeters: String? = null,
    val stops: List<RouteStopDto>,
    val path: List<GeoPointDto> = emptyList(),
)
